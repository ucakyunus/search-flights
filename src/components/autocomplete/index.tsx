import React, { forwardRef } from 'react';
import {
  Autocomplete as MUIAutocomplete,
  AutocompleteProps as MUIAutocompleteProps,
  InputAdornment,
  TextField,
} from '@mui/material';
import { ReturnLocation } from '@/types/location';

export interface AutocompleteProps
  extends Omit<MUIAutocompleteProps<ReturnLocation, false, false, false>, 'renderInput'> {
  startAdornment?: React.ReactNode;
  placeholder?: string;
}

export const Autocomplete = forwardRef(
  ({ placeholder, startAdornment, ...rest }: AutocompleteProps, forwardedRef) => {
    return (
      <MUIAutocomplete
        clearIcon={null}
        noOptionsText='Sonuç bulunamadı'
        sx={{ width: '100%', bgcolor: 'white' }}
        getOptionLabel={(option) => option.label}
        getOptionKey={(option) => option.value}
        renderInput={(params) => (
          <div style={{ height: '100%' }}>
            <TextField
              {...params}
              sx={{
                height: '100%',
                '.MuiOutlinedInput-root': {
                  height: '100%',
                },
              }}
              placeholder={placeholder}
              variant='outlined'
              inputRef={forwardedRef}
              slotProps={{
                input: {
                  ...params.InputProps,
                  startAdornment: (
                    <InputAdornment position='start' sx={{ color: 'primary.dark' }}>
                      {startAdornment}
                    </InputAdornment>
                  ),
                  sx: {
                    fontSize: '1.3rem',
                  },
                },
              }}
              fullWidth
            />
          </div>
        )}
        {...rest}
      />
    );
  },
);
