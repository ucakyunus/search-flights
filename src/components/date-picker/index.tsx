import { IconButton, IconButtonProps, Typography } from '@mui/material';
import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
import { styled } from '@mui/material/styles';

const StyledIconButton = styled(IconButton)<IconButtonProps>(({ theme }) => ({
  backgroundColor: theme.palette.secondary.dark,
  color: 'white',
  borderRadius: 0,
  width: 90,
  border: `1px solid ${theme.palette.secondary.dark}`,
  '&:hover': {
    backgroundColor: theme.palette.primary.main,
    border: `1px solid ${theme.palette.common.white}`,
  },
}));

export const DatePicker = () => {
  return (
    <StyledIconButton size={'large'}>
      <Typography fontSize={14} mr={1.5}>
        Tarih
      </Typography>
      <CalendarMonthIcon fontSize='large' />
    </StyledIconButton>
  );
};
