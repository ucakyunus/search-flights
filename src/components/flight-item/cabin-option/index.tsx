import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { Box, Radio, Button, RadioProps, Typography, ButtonProps, useTheme } from '@mui/material';
import styled from '@emotion/styled';

export interface CabinOptionProps extends ButtonProps {
  price: number;
  label: string;
  currency: string;
  radioProps?: RadioProps;
  isSelected?: boolean;
}

export const CabinOption = ({
  price,
  label,
  currency,
  radioProps,
  isSelected = false,
  ...rest
}: CabinOptionProps) => {
  const { custom } = useTheme();

  const StyledButton = styled(Button)({
    boxShadow: custom?.shadows![1],
    width: 200,
    justifyContent: 'flex-start',
    backgroundColor: 'white',
    ...(isSelected && {
      ':after': {
        content: '""',
        position: 'absolute',
        bottom: -16,
        left: 0,
        width: '100%',
        height: 16,
        backgroundColor: 'white',
      },
      ':hover': {
        backgroundColor: 'white',
      },
    }),
  });

  return (
    <StyledButton {...rest}>
      <Radio {...radioProps} />

      <Typography
        ml={-0.5}
        color='gray'
        fontSize={8}
        fontWeight={600}
        sx={{ textDecoration: 'underline' }}
      >
        {label}
      </Typography>

      <Box display='flex' flexDirection='column' alignItems='start' ml={1}>
        <Typography fontSize={8} color='gray'>
          Yolcu Basina
        </Typography>
        <Typography fontSize={14} fontWeight={600}>
          {currency} {price}
        </Typography>
      </Box>

      <ExpandMoreIcon sx={{ marginLeft: 'auto' }} />
    </StyledButton>
  );
};
