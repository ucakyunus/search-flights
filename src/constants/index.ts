export enum Currency {
  TRY = 'TRY',
}

export enum FareCategories {
  BUSINESS = 'BUSINESS',
  ECONOMY = 'ECONOMY',
}

export enum BrandCode {
  ECO_FLY = 'ecoFly',
  EXTRA_FLY = 'extraFly',
  PRIME_FLY = 'primeFly',
}

export enum Status {
  AVAILABLE = 'AVAILABLE',
  ERROR = 'ERROR',
}

export enum ListSortTypes {
  PriceAscending = 'PriceAscending',
  PriceDescending = 'PriceDescending',
  DepartureTimeAscending = 'DepartureTimeAscending',
  DepartureTimeDescending = 'DepartureTimeDescending',
}

export enum ListSortBy {
  DepartureTime = 'DepartureTime',
  Price = 'Price',
}

export enum ResultTypes {
  Success = 'Success',
  Failure = 'Failure',
}

export enum CounterActions {
  Increment = 'INCREMENT',
  Decrement = 'DECREMENT',
}
