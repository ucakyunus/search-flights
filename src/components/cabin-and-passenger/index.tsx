import React, { useMemo, useState } from 'react';
import { IconButton, Typography } from '@mui/material';
import ManIcon from '@mui/icons-material/Man';
import AddIcon from '@mui/icons-material/Add';

import { CabinSelectorPopover } from '@/components/cabin-and-passenger/cabin-selector-popover';
import { FareCategories } from '@/constants';

export interface CabinAndPassengerProps {
  onPersonCount: (count: number) => void;
  onCabinType: (cabinType: FareCategories) => void;
  personCount: number;
  cabinType: FareCategories;
}

export const CabinAndPassenger = ({
  personCount,
  cabinType,
  onPersonCount,
  onCabinType,
}: CabinAndPassengerProps) => {
  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const icon = useMemo(() => {
    if (personCount > 3) {
      return (
        <div style={{ display: 'flex', alignItems: 'center' }}>
          {[...Array(3)].map((_, index) => (
            <ManIcon key={index} sx={{ width: 20 }} />
          ))}
          <AddIcon fontSize='small' sx={{ width: 20 }} />
        </div>
      );
    }

    return (
      <div style={{ display: 'flex' }}>
        {[...Array(personCount)].map((_, index) => (
          <ManIcon key={index} />
        ))}
      </div>
    );
  }, [personCount]);

  return (
    <>
      <IconButton
        onClick={handleClick}
        sx={{
          backgroundColor: 'secondary.dark',
          color: 'white',
          borderRadius: 0,
          width: 90,
          border: `1px solid ${'secondary.dark'}`,
          '&:hover': {
            backgroundColor: 'primary.main',
            border: `1px solid ${'common.white'}`,
          },
        }}
      >
        {icon}
        <Typography py={0.3} px={0.6} fontSize={14} position='absolute' top={0} right={0}>
          {personCount}
        </Typography>
      </IconButton>

      {anchorEl ? (
        <CabinSelectorPopover
          anchorEl={anchorEl}
          onClose={handleClose}
          cabinType={cabinType}
          personCount={personCount}
          onCabinType={onCabinType}
          onPersonCount={onPersonCount}
        />
      ) : null}
    </>
  );
};
