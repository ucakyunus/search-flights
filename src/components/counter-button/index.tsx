import { Box, IconButton, IconButtonProps, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';
import { Remove, Add } from '@mui/icons-material';

import { CounterActions } from "@/constants";

export interface CounterButtonProps {
  count: number;
  min?: number;
  max?: number;
  onChange: (count: number) => void;
}

const IncrementButton = styled(IconButton)<IconButtonProps>(({ theme }) => ({
  color: theme.palette.secondary.dark,
  width: 30,
  height: 30,
  padding: 0,
  borderRadius: 4
}));

export const CounterButton = ({ count, min = 1, max = 10, onChange }: CounterButtonProps) => {
  
  const handleCountChange = (action: CounterActions) => {
    const newCount =
      action === CounterActions.Increment ? count + 1 : count - 1;
    
    if (newCount >= min && newCount <= max) {
      onChange(newCount);
    }
  };
  
  return (
    <Box display="flex">
      <IncrementButton
        onClick={() => handleCountChange(CounterActions.Decrement)}
        sx={{
          bgcolor: count === min ? 'ButtonShadow' : 'secondary.light'
        }}
      >
        <Remove color="inherit" sx={{ fontSize: 20 }} />
      </IncrementButton>
      
      <Typography fontSize={14} px={2} lineHeight={2}>
        {count}
      </Typography>
      
      <IncrementButton
        onClick={() => handleCountChange(CounterActions.Increment)}
        sx={{
          bgcolor: count === max ? 'ButtonShadow' : 'secondary.light'
        }}
      >
        <Add color="inherit" sx={{ fontSize: 20 }} />
      </IncrementButton>
    </Box>
  );
};