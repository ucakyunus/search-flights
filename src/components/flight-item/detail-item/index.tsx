import { useNavigate } from 'react-router-dom';
import { Box, Button, Typography } from '@mui/material';

import { BrandCode, ResultTypes, Status } from '@/constants';

import type { FareSubcategory } from '@/types/flight';

export interface DetailItemProps {
  detail: FareSubcategory;
  isPromotionActive: boolean;
}

export const DetailItem = ({ detail, isPromotionActive }: DetailItemProps) => {
  const { status, price, brandCode, rights } = detail;

  const navigate = useNavigate();

  const isAvailable = status === Status.AVAILABLE;
  const isEcoFly = brandCode === BrandCode.ECO_FLY;

  const isButtonDisabled = isPromotionActive && !isEcoFly;

  const newPrice = isEcoFly && isPromotionActive ? price.amount / 2 : price.amount;

  const onSelectHandler = () => {
    const state = isAvailable
      ? {
          status: ResultTypes.Success,
          amount: newPrice,
          currency: price.currency,
        }
      : { status: ResultTypes.Failure };

    navigate('/result', { state });
  };

  return (
    <Box
      width={400}
      display='flex'
      minHeight={300}
      alignItems='stretch'
      flexDirection='column'
      sx={{ backgroundColor: 'white' }}
    >
      <Box
        p={1}
        height={60}
        display='flex'
        alignItems='flex-start'
        bgcolor='secondary.light'
        justifyContent='space-between'
      >
        <Typography fontSize={16} fontWeight={600} textTransform='capitalize'>
          {brandCode}
        </Typography>

        <Box display={'flex'} gap={1}>
          {isPromotionActive && isEcoFly && (
            <Box
              display='flex'
              alignItems='flex-start'
              sx={{ textDecoration: 'line-through', opacity: 0.4 }}
            >
              <Typography fontSize={12}>{price.currency}</Typography>
              <Typography fontSize={16} ml={0.5}>
                {price.amount}
              </Typography>
            </Box>
          )}
          <Box display='flex' alignItems='flex-start'>
            <Typography fontSize={12}>{price.currency} </Typography>
            <Typography fontSize={16} ml={0.5}>
              {newPrice}
            </Typography>
          </Box>
        </Box>
      </Box>
      <Box flex={1} border={1} borderColor='secondary.light'>
        {rights.map((right, index) => (
          <Typography
            p={1}
            key={index}
            fontSize={12}
            borderBottom={1}
            borderColor='secondary.light'
          >
            {right}
          </Typography>
        ))}
      </Box>
      <Button
        fullWidth
        size='large'
        color='error'
        variant='contained'
        onClick={onSelectHandler}
        disabled={isButtonDisabled}
      >
        Uçuşu Seç
      </Button>
    </Box>
  );
};
