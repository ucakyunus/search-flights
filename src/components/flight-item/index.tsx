import { useMemo } from 'react';
import { Box, Divider, useTheme } from '@mui/material';

import { CabinOption } from '@/components/flight-item/cabin-option';
import { InfoDetail } from '@/components/flight-item/info-detail';
import { Duration } from '@/components/flight-item/duration';
import { DetailItem } from '@/components/flight-item/detail-item';
import { FareCategories } from '@/constants';

import type { Flight } from '@/types/flight';

export interface FlightItemProps {
  index: number;
  data: Flight;
  isPromotionActive: boolean;
  selectedFareOption: string | null;
  setSelectedFareOption: (value: string) => void;
}

export const FlightItem = (props: FlightItemProps) => {
  const { custom } = useTheme();

  const { data, index, isPromotionActive, selectedFareOption, setSelectedFareOption } = props;

  const { fareCategories } = data;

  const businessRadioValue = `${FareCategories.BUSINESS}-${index}`;
  const economyRadioValue = `${FareCategories.ECONOMY}-${index}`;
  const isBusinessSelected = selectedFareOption === businessRadioValue;
  const isEconomySelected = selectedFareOption === economyRadioValue;

  const flightItemDetailItems = useMemo(() => {
    if (!selectedFareOption) return [];

    if (selectedFareOption !== businessRadioValue && selectedFareOption !== economyRadioValue)
      return [];

    const [selectedCabinType] = selectedFareOption.split('-');

    const subcategories =
      selectedCabinType === FareCategories.BUSINESS
        ? fareCategories.BUSINESS.subcategories
        : fareCategories.ECONOMY.subcategories;

    return subcategories.sort((a, b) => a.order - b.order);
  }, [
    businessRadioValue,
    economyRadioValue,
    fareCategories.BUSINESS.subcategories,
    fareCategories.ECONOMY.subcategories,
    selectedFareOption,
  ]);

  const businessEcoFlyPrice = useMemo(
    () =>
      fareCategories.BUSINESS.subcategories.find((category) => {
        return category.order === 1;
      })?.price.amount as number,
    [fareCategories.BUSINESS.subcategories],
  );

  const economyEcoFlyPrice = useMemo(
    () =>
      fareCategories.ECONOMY.subcategories.find((category) => {
        return category.order === 1;
      })?.price.amount as number,
    [fareCategories.ECONOMY.subcategories],
  );

  const currency = useMemo(
    () => fareCategories.BUSINESS.subcategories[0].price.currency,
    [fareCategories.BUSINESS.subcategories],
  );

  const economyPriceShown = useMemo(() => {
    return isPromotionActive ? economyEcoFlyPrice / 2 : economyEcoFlyPrice;
  }, [economyEcoFlyPrice, isPromotionActive]);

  const businessPriceShown = useMemo(() => {
    return isPromotionActive ? businessEcoFlyPrice / 2 : businessEcoFlyPrice;
  }, [businessEcoFlyPrice, isPromotionActive]);

  return (
    <Box>
      <Box display='flex' gap={2}>
        <Box
          p={2}
          flex={1}
          display='flex'
          alignItems='center'
          justifyContent='space-between'
          boxShadow={custom?.shadows![1]}
          sx={{ backgroundColor: 'white' }}
        >
          <Box display='flex' alignItems='center' flex={1}>
            <InfoDetail
              time={data.departureDateTimeDisplay}
              portCode={data.originAirport.code}
              cityName={data.originAirport.city.name}
            />

            <Divider
              sx={{
                width: '400px',
                height: '1px',
                backgroundColor: '#000',
                mx: 1,
              }}
            />

            <InfoDetail
              textAlign='right'
              time={data.arrivalDateTimeDisplay}
              portCode={data.destinationAirport.code}
              cityName={data.destinationAirport.city.name}
            />
          </Box>

          <Duration mr={3} time={data.flightDuration} />
        </Box>

        <CabinOption
          label='Economy'
          currency={currency}
          price={economyPriceShown}
          isSelected={isEconomySelected}
          onClick={() => setSelectedFareOption(economyRadioValue)}
          radioProps={{
            value: economyRadioValue,
            checked: isEconomySelected,
            onChange: () => setSelectedFareOption(economyRadioValue),
          }}
        />

        <CabinOption
          label='Business'
          currency={currency}
          price={businessPriceShown}
          isSelected={isBusinessSelected}
          onClick={() => setSelectedFareOption(businessRadioValue)}
          radioProps={{
            value: businessRadioValue,
            checked: isBusinessSelected,
            onChange: () => setSelectedFareOption(businessRadioValue),
          }}
        />
      </Box>

      {flightItemDetailItems.length > 0 ? (
        <Box display='flex' gap={2} p={2} mt={2} bgcolor={'white'}>
          {flightItemDetailItems.map((item, index) => (
            <DetailItem key={index} detail={item} isPromotionActive={isPromotionActive} />
          ))}
        </Box>
      ) : null}
    </Box>
  );
};
