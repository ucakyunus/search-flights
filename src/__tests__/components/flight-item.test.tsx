import { render } from '@testing-library/react';

import { FlightItem, FlightItemProps } from '@/components/flight-item';
import { BrandCode, Currency, Status } from '@/constants';

const mockProps: FlightItemProps = {
  data: {
    departureDateTimeDisplay: '10:00 AM',
    arrivalDateTimeDisplay: '12:30 PM',
    originAirport: {
      name: 'Istanbul Airport',
      code: 'IST',
      city: { code: 'IST', name: 'Istanbul' },
      country: { code: 'TR', name: 'Turkey' },
    },
    destinationAirport: {
      name: 'Antalya Airport',
      code: 'AYT',
      city: { code: 'AYT', name: 'Antalya' },
      country: { code: 'TR', name: 'Turkey' },
    },
    flightDuration: '5h 30m',
    fareCategories: {
      BUSINESS: {
        subcategories: [
          {
            order: 1,
            price: { amount: 400, currency: Currency.TRY },
            brandCode: BrandCode.ECO_FLY,
            status: Status.AVAILABLE,
            rights: ['20 kg Bagaj', 'Ucresiz Yemek Secimi'],
          },
          {
            order: 2,
            price: { amount: 500, currency: Currency.TRY },
            brandCode: BrandCode.EXTRA_FLY,
            status: Status.AVAILABLE,
            rights: ['30 kg Bagaj', 'Standart Koltuk Secimi', 'Ucresiz Yemek Secimi'],
          },
          {
            order: 3,
            price: { amount: 1000, currency: Currency.TRY },
            brandCode: BrandCode.PRIME_FLY,
            status: Status.AVAILABLE,
            rights: [
              '50 kg Bagaj',
              'Standart Koltuk Secimi',
              'Ucretsiz Degisiklik',
              'Ucresiz Yemek Secimi',
            ],
          },
        ],
      },
      ECONOMY: {
        subcategories: [
          {
            order: 1,
            price: { amount: 200, currency: Currency.TRY },
            brandCode: BrandCode.ECO_FLY,
            status: Status.AVAILABLE,
            rights: ['15 kg Bagaj'],
          },
          {
            order: 2,
            price: { amount: 400, currency: Currency.TRY },
            brandCode: BrandCode.EXTRA_FLY,
            status: Status.AVAILABLE,
            rights: ['20 kg Bagaj', 'Standart Koltuk Secimi'],
          },
          {
            order: 1,
            price: { amount: 800, currency: Currency.TRY },
            brandCode: BrandCode.PRIME_FLY,
            status: Status.AVAILABLE,
            rights: ['25 kg Bagaj', 'Standart Koltuk Secimi', 'Ucretsiz Degisiklik'],
          },
        ],
      },
    },
  },
  index: 0,
  isPromotionActive: true,
  selectedFareOption: null,
  setSelectedFareOption: jest.fn(),
};

test('FlightItem matches snapshot', () => {
  const { asFragment } = render(<FlightItem {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
