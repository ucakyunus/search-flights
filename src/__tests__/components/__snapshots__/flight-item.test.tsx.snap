// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightItem matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-10egq61"
    >
      <div
        class="MuiBox-root css-1sd21uz"
      >
        <div
          class="MuiBox-root css-17rqx3h"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-3kdse5-MuiTypography-root"
            >
              10:00 AM
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-fp2ujc-MuiTypography-root"
            >
              IST
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-14s8yo4-MuiTypography-root"
            >
              Istanbul
            </p>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-35cizu-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-s2uf1z"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-3kdse5-MuiTypography-root"
            >
              12:30 PM
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-fp2ujc-MuiTypography-root"
            >
              AYT
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-14s8yo4-MuiTypography-root"
            >
              Antalya
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root css-1n8brx5"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-o6dx9f-MuiTypography-root"
          >
            Uçuş Süresi
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-izap9d-MuiTypography-root"
          >
            5h 30m
          </p>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1ak7fun-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            type="radio"
            value="ECONOMY-0"
          />
          <span
            class="css-1qiat4j"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1luxhcg-MuiTypography-root"
        >
          Economy
        </p>
        <div
          class="MuiBox-root css-aqbzds"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-mkiahx-MuiTypography-root"
          >
            Yolcu Basina
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-3kdse5-MuiTypography-root"
          >
            TRY 100
          </p>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-8n04zg-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1ak7fun-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-13lvt8g-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-j8yymo"
            type="radio"
            value="BUSINESS-0"
          />
          <span
            class="css-1qiat4j"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1bz1rr0-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-o8rdeq-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1luxhcg-MuiTypography-root"
        >
          Business
        </p>
        <div
          class="MuiBox-root css-aqbzds"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-mkiahx-MuiTypography-root"
          >
            Yolcu Basina
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-3kdse5-MuiTypography-root"
          >
            TRY 200
          </p>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-8n04zg-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
